src/plugins/formulas/formulas.js

```diff
@@ -1,425 +1,197 @@
+   * @param {Number} row Row index.
    * @param {Number} column Column index.
-   * @param {Number} amount An amount of created columns.
+   * @param {Object} valueHolder Value holder as an object to change value by reference.
+   * @param {String} ioMode IO operation (`get` or `set`).
+   * @returns {Array|undefined} Returns modified row data.
    */
-  onAfterRemoveCol(column, amount) {
-    this.sheet.alterManager.triggerAlter('remove_column', column, amount);
+  onModifyData(row, column, valueHolder, ioMode) {
+    if (this.isEnabled()) {
+      if (ioMode === 'get') {
+        valueHolder.value = this.hyperformula.getCellValue({
+          row,
+          col: column,
+          sheet: this.sheetId
+        });
+
+        valueHolder.value = parseHFValue(valueHolder.value);
+
+      } else if (ioMode === 'set') {
+        this.hyperformula.setCellContent({
+          row,
+          col: column,
+          sheet: this.sheetId
+        }, valueHolder.value);
+      }
+    }
   }
 
   /**
-   * On before column sorting listener.
-   *
-   * @private
-   * @param {Number} column Sorted column index.
-   * @param {Boolean} order Order type.
+   * TODO: docs + correct the modifySourceData hook
+   * @param row
+   * @param column
+   * @param valueHolder
+   * @param ioMode
    */
-  onBeforeColumnSort(column, order) {
-    this.sheet.alterManager.prepareAlter('column_sorting', column, order);
+  onModifySourceData(row, column, valueHolder, ioMode) {
+    const sheetDimensions = this.hyperformula.getSheetDimensions(this.sheetId);
+
+    if (this.isEnabled() && (sheetDimensions.width !== 0 && sheetDimensions.height !== 0)) {
+      if (ioMode === 'get') {
+        valueHolder.value = this.hyperformula.getCellFormula({
+          row,
+          col: column,
+          sheet: this.sheetId
+        }) || this.hyperformula.getCellValue({
+          row,
+          col: column,
+          sheet: this.sheetId
+        }); //TODO: optimize
+
+        valueHolder.value = parseHFValue(valueHolder.value);
+      }
+    }
   }
 
   /**
-   * On after column sorting listener.
+   * On after column sorting listener. TODO: docs
    *
    * @private
    * @param {Number} column Sorted column index.
    * @param {Boolean} order Order type.
    */
-  onAfterColumnSort(column, order) {
-    this.sheet.alterManager.triggerAlter('column_sorting', column, order);
+  onAfterColumnSort(currentSortConfig, destinationSortConfigs) {
+    // TODO: no sorting implementation in hyperformula yet
   }
 
   /**
-   * On after load data listener.
+   * Checks if the plugin is enabled in the handsontable settings. This method is executed in {@link Hooks#beforeInit}
+   * hook and if it returns `true` than the {@link HyperFormula#enablePlugin} method is called.
    *
-   * @private
-   */
-  onAfterLoadData() {
-    this._skipRendering = true;
-    this.recalculateFull();
-  }
-
-  /**
-   * Destroys the plugin instance.
+   * @returns {Boolean}
    */
-  destroy() {
```

Comment:
Add `destroy` as a TODO method. We should fix this as soon as https://github.com/handsontable/hyperformula/issues/45 is done.